src/components/Articles.jsx

```diff
@@ -2,14 +2,16 @@ import { useContext, useEffect, useState } from "react";
 import { ArticleContext } from "../contexts/Article";
 import { getArticles } from "../api";
 import ArticleCard from "./ArticleCard";
+import { useParams } from "react-router-dom";
 
 const Articles = () => {
   const { articles, setArticles } = useContext(ArticleContext);
   const [isLoading, setIsLoading] = useState(true);
+  const { slug } = useParams();
```

Comment:
Nice implementation of a parametric endpoint here
I also like that instead of making a new component for a list of articles by topic, you've made the existing articles component more flexible 👏